systemtest/src/main/java/io/strimzi/systemtest/performance/utils/TopicOperatorPerformanceUtils.java

```diff
@@ -0,0 +1,299 @@
+     * │   batches, ensuring robust execution.                                                            │
+     * └──────────────────────────────────────────────────────────────────────────────────────────────────┘
+     *
+     * @param testStorage    The storage instance containing the configuration and namespace details.
+     * @param numberOfTopics The total number of Kafka topics to process.
+     */
+    public static void processKafkaTopicBatchesAsync(TestStorage testStorage, int numberOfTopics) {
+        final int topicsPerBatch = (numberOfTopics + AVAILABLE_CPUS - 1) / AVAILABLE_CPUS; // Ensures all topics are covered
+
+        final ExtensionContext currentContext = ResourceManager.getTestContext();
+        final CompletableFuture<?>[] futures = new CompletableFuture[AVAILABLE_CPUS];
+
+        for (int batch = 0; batch < AVAILABLE_CPUS; batch++) {
+            int start = batch * topicsPerBatch;
+            int end = Math.min(start + topicsPerBatch, numberOfTopics); // Ensure we do not go out of bounds
+            // Delegate batch processing to a separate method
+            futures[batch] = processBatch(start, end, currentContext, testStorage);
+        }
+        CompletableFuture.allOf(futures).join(); // Wait for all batches to complete
+        LOGGER.info("All batches completed.");
+    }
+
+    /**
+     * Manages the creation, modification, and deletion of Kafka topics for a specific batch.
+     * Operations are performed sequentially within each batch but executed concurrently across multiple batches,
+     * utilizing multiple threads to improve performance.
+     *
+
+     *
+     * @param start           Starting index of the batch.
+     * @param end             Ending index of the batch.
+     * @param currentContext  The current JUnit extension context.
+     * @param testStorage     Storage instance with namespace and configuration details.
+     * @return                CompletableFuture representing the completion of all operations in the batch.
+     */
+    private static CompletableFuture<Void> processBatch(int start, int end, ExtensionContext currentContext,
+                                                        TestStorage testStorage) {
+        return CompletableFuture.runAsync(() -> performCreationWithWait(start, end, currentContext, testStorage), EXECUTOR)
+            .thenRunAsync(() -> performModificationWithWait(start, end, currentContext, testStorage, KAFKA_TOPIC_CONFIG_TO_MODIFY), EXECUTOR)
+            .thenRunAsync(() -> performDeletionWithWait(start, end, currentContext, testStorage), EXECUTOR)
+            .exceptionally(ex -> {
+                LOGGER.error("Error processing batch from {} to {}: {}", start, end, ex.getMessage(), ex);
+                return null;
+            });
+    }
+
+    private static void performCreationWithWait(int start, int end, ExtensionContext currentContext, TestStorage testStorage) {
+        ResourceManager.setTestContext(currentContext);
```

Comment:
Why the test context is set here, isn't it something that RM should handle by itself?

Reply:
Because this method is invoked in the new `Thread`. Therefore if you do not set context you would end up in the NPE because a new Thread does not hold the state of the `ExtensionContext`, and so you need to set it.

Reply:
Okay, I would add a comment about this there, why it's needed. (I know that for you it makes sense, but when someone else will go through the code, it will be similarly confusing for him as it was for me)